internal/cache/cache.go

```diff
@@ -208,9 +210,17 @@ func New(ctx context.Context, opts ...Option) (c *Cache, err error) {
 
 	logger.Debug(ctx, "Shadow db mode: %v", shadowMode)
 
-	if shadowMode == shadowRWMode {
-		offlineCredentialsExpirationDuration := time.Duration(2 * uint64(o.offlineCredentialsExpiration) * 24 * uint64(time.Hour))
-		if err := cleanUpDB(ctx, db, offlineCredentialsExpirationDuration); err != nil {
+	if shadowMode == shadowRWMode && o.offlineCredentialsExpiration != 0 {
+		d := o.offlineCredentialsExpiration
+		if o.offlineCredentialsExpiration < 0 {
+			d = defaultCredentialsExpiration
+		}
+		days := uint64(d) * expirationPurgeModifier
+
+		logger.Debug(ctx, "Cleaning up db. Removing entries that last authenticated online more than %d days ago", days)
+
+		maxCacheEntryDuration := time.Duration(days * 24 * uint64(time.Hour))
+		if err := cleanUpDB(ctx, db, maxCacheEntryDuration); err != nil {
 			return nil, err
 		}
```

Comment:
I would put an else here for debug telling that we don’t clean up the database because offlineCredentialsExpiration is that to 0, as the one you removed in `CleanUpDB`.